### PR TITLE
Character sheet now retains and shows library data for traits and abi…

### PR DESCRIPTION
**Pull request by baoghal**
…liites.

1. Links on traits and abilities now show full edit window when clicked.
2. Skill level calculation now handles estimated defaults and point values of "3"
3. Skill level listed is based on points.
4. Traits and abilities, if on a list already, wonlt allow you to change the type to a type that doesn't belong in that list (i.e. changing a skill on the list to a spell is undone)

### CODE REVIEW

File: scripts/manager_actor2.lua [hunk 1]
```diff
@@ -221,7 +237,11 @@ function getSkillLevel(nodeChar, type, points)
 		end
 	end
 
-	return nLevel, skilltype, relativelevel;
```
Comment (baoghal) [inline scripts/manager_actor2.lua:224]: There seems to be a bug with FGU where this triplet was not always returning all values. This seems to work around that problem altogether. Like, I was using chat messages to see that all three value were valid, and then immediately after the function call some of them would vanish.

**Comment by rinickolous** [comment]
>1. Links on traits and abilities now show full edit window when clicked.

This works for abilities, but it doesn't seem to work for traits.
See attached:
image
The flyout window on the right appears when clicking the link in the traits library, whereas the one on the left appears when clicking the link in the actual character sheet.

Perhaps I misunderstood the notes here, but is this intentional?

**Comment by baoghal** [comment]
The initial commit added it for traits, buy I backed that out on the
subsequent commit. Sorry I didn't make that clear.

**Comment by rinickolous** [comment]
Thanks. What's the reason for undoing that change?

**Comment by baoghal** [comment]
Tldr: it's coming back soon. :)

I pulled out all of my changes based on ronnke's initial feedback. On my
fork, this is once again a pop-up window, but it isn't a trait window, it
is more of a "trait instance" window. Once I get through with other things
that were asked for, I'll be making another pull request to this branch
that includes the pop-up window for traits once again.
